src/main/java/com/stripe/model/PaymentIntentSourceActionDeserializer.java

```diff
@@ -29,13 +33,29 @@
   public PaymentIntentSourceAction deserialize(JsonElement json, Type typeOfT,
       JsonDeserializationContext context)
       throws JsonParseException {
-    PaymentIntentSourceAction sourceAction = new PaymentIntentSourceAction();
-    JsonObject jsonObject = json.getAsJsonObject();
-    String type = jsonObject.get("type").getAsString();
-    sourceAction.setType(type);
+    if (json.isJsonNull()) {
+      return null;
+    }
+
+    if (!json.isJsonObject()) {
+      throw new JsonParseException(
+          "Next source action was not an object, which is problematic.");
+    }
+
+    JsonObject actionAsJsonObject = json.getAsJsonObject();
+    JsonElement valueJson = actionAsJsonObject.get("value");
+    actionAsJsonObject.remove("value");
+
+    Gson gson = new GsonBuilder()
+        .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+        .create();
+    PaymentIntentSourceAction sourceAction = gson.fromJson(json, PaymentIntentSourceAction.class);
```

Comment:
I mostly used what we do for the BT deserialization but not sure it's the right approach.